### DISCUSSION TITLE
Using autodoc to generate documentation for downstream project

### DISCUSSION THREAD

**Discussion by thomasrutger** [eclipse-edc/GradlePlugins#290] [Q&A] [answered]
I am trying to use the autodoc extension to generate documentation for my downstream project (which has a similar structure as the template project: https://github.com/eclipse-edc/Template-Basic). I have some runtimes together with custom extensions. 

Basically I would like to generate autodoc manifests for my runtimes, which should include both the manifests from my own custom extensions as well as any manifests from EDC modules that the runtimes depend on, so that I can create markdown files using `doc2md`  that cover all the configuration settings documentation for the entire runtime. 

My assumption is that if I execute the `downloadManifests` task, which should pull in any manifest file that my project depends on (both custom modules and my own modules), followed by `mergeManifests` it should achieve what I want. However, when I run `./gradlew :runtimes:control-plane:downloadManifests`, I only see the json manifests from my own extensions in `runtimes/control-plane/build/autodoc`, but none of the manifests of the EDC modules that my runtime depends on. Therefore my question is:

1. Does the autodoc plugin support my use case (i.e. is the above assumption valid)?
2. and if it does, how can I achieve it?

I am using version 0.10.0 of the EDC-build plugin. I have read https://eclipse-edc.github.io/documentation/for-contributors/autodoc/, but this did not help.

**Comment by paullatzelsperger** [comment]
try this: in your root project execute:
```shell
./gradlew autodoc # to generate the autodoc manifest in every of your modules
./gradlew downloadManifests # to obtain autodoc manifests from upstream EDC modules. They are available as Maven artifacts
./gradlew mergeManifests # to merge them with your own modules
```

this should put the merged autodoc manifest in `build/manifest.json`

**Reply by thomasrutger** [reply]
I tried this (making sure to run ./gradlew clean beforehand), but it does not achieve the desired result. None of the upstream EDC manifests are included in the endresult (build/manifest.json). It just contains the merged `edc.json` files for my own custom modules (excluding the runtime modules somehow, these don´t seem to produce `edc.json` files, perhaps because they do not contain code).

**Reply by paullatzelsperger** [reply, marked as answer]
this exact use case worked, at least at some point in the past. we did recently refactor the plugins a bit, and it is always possible, that something internal to the Gradle API changed. In any event, i'll create a bug issue for this, because something clearly isn't working here.

Thanks for raising!
